app/Http/Middleware/RequestCost.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+
+class RequestCost
+{
+    public static function getCost(Request $request): int
+    {
+        return $request->attributes->get('request_cost', 1);
+    }
+
+    public static function setCost(int $cost, ?Request $request = null)
```

Comment:
`$request` is optional as convenience so controllers can just call `RequestCost::setCost(x)` instead of `RequestCost::setCost(request(), x)`.
It doesn't make sense to use `request()` in `handle` when `$request` is there as a parameter.